modules/sor/sorV2/lib/poolsV3/stable/stablePool.ts

```diff
@@ -156,6 +179,13 @@ export class StablePool implements BasePoolV3 {
         let calculatedAmount: bigint;
 
         if (tIn.token.isSameAddress(this.id)) {
+            // if liquidityManagement.disableUnbalancedLiquidity is true return 0
+            // as the pool does not allow unbalanced operations. 0 return marks the
+            // route as truly unfeasible route.
+            if (this.liquidityManagement.disableUnbalancedLiquidity) {
+                return TokenAmount.fromRawAmount(tOut.token, 0n);
+            }
```

Comment:
Just sharing that I talked to @johngrantuk and he feels this is better suited here in the backend code instead of balancer-maths. Let's leave as is 😉

Reply:
@brunoguerios I canno't find the original chat message but I recall you talking to john and agreeing that this can be left in here instead of moving it to balancer-maths.